au/code/au/utility/string_constant.hh

```diff
@@ -87,6 +90,22 @@ struct IToA;
 // Implementation details below.
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+// The absolute value of a signed integer, as an unsigned integer.
+//
+// This handles the special case where the lowest `int64_t` cannot be directly negated: it would be
+// too big to fit into `int64_t`.
+template <typename S>
+constexpr std::make_unsigned_t<S> abs_as_unsigned(S x) {
+    static_assert(stdx::conjunction<std::is_integral<S>, std::is_signed<S>>::value,
+                  "Only designed for signed integral types");
+    using U = std::make_unsigned_t<S>;
+    constexpr auto SMAX = static_cast<U>(std::numeric_limits<S>::max());
+    constexpr auto UMAX = std::numeric_limits<U>::max();
+
+    auto result = static_cast<U>(x);
+    return (result > SMAX) ? (UMAX - result + 1u) : result;
```

Comment:
I don't need the parens for readability (years of C I've learned ternary is pretty far down the precedence list, only above comma), but understand if you want to leave them in.

Reply:
Yep: for ternary specifically, I never really feel uncertainty about precedence.  However, I do like the visual grouping that the parentheses bring.